### PR TITLE
Tiny documentation enhancements

### PR DESCRIPTION
**Pull request by Totktonada**
- Added forgotten `crud-router` mention.
- Added tables of contents (using [doctoc](https://github.com/thlorenz/doctoc)).
- Added Vim Swap files into .gitignore.

### CODE REVIEW

**Review by olegrok — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,6 @@ luacov.report.out
 luacov.stats.out
 build/*.cmake
 build/Makefile
+
+# Vim Swap files.
+.*.sw[a-z]
```
Comment (oleg-jukovec) [inline .gitignore:15]: http://vimdoc.sourceforge.net/htmldoc/recover.html#swap-file
```
If the ".swp" file name already exists, the last character is
	   decremented until there is no file with that name or ".saa" is
	   reached.  In the last [case](http://vimdoc.sourceforge.net/htmldoc/change.html#case), no swap file is created.
```
```suggestion
.*.s[a-w][a-z]
```

**Review by oleg-jukovec — APPROVED** [review]
Thanks for the pull request. I like it, but you can improve the regular expression for vim swap files if you want.

File: .gitignore (hunk 1 above)
Comment (Totktonada) [reply .gitignore]: No problem, updated.

**Review by DifferentialOrange — APPROVED** [review]
I think two ToC commits could be merged into one, but it's a matter of taste

**Pull request merged**